.semgrep.yml

```diff
@@ -151,6 +151,27 @@ rules:
         - pattern: '*aws.Time($VALUE)'
     severity: WARNING
 
+  - id: data-source-with-resource-read
+    languages: [go]
+    message: Calling a resource's Read method from within a data-source is discouraged
+    paths:
+      include:
+        - aws/data_source_aws_*.go
+    patterns:
+      - pattern: $RESOURCE_READ
```

Comment:
Nit: Can use [`pattern-regex`](https://semgrep.dev/docs/writing-rules/rule-syntax/#pattern-regex) instead of `pattern`+`metavariable-regex` in this situation